js/ui/appDisplay.js

```diff
@@ -2003,19 +2161,20 @@ const FolderIcon = new Lang.Class({
     Name: 'FolderIcon',
     Extends: ViewIcon,
 
-    _init: function(id, parentView) {
+    _init: function(dirInfo, parentView) {
         let viewIconParams = { isDraggable: true,
                                parentView: parentView };
         let buttonParams = { button_mask: St.ButtonMask.ONE,
                              toggle_mode: true };
         let iconParams = { createIcon: Lang.bind(this, this._createIcon),
                            setSizeManually: false,
                            editable: true };
-        this.id = id;
+
         this.name = '';
         this._parentView = parentView;
 
-        this._dirInfo = Shell.DesktopDirInfo.new(id);
+        this.id = dirInfo.get_id();
```

Comment:
We already have `getId()` methods on the icons, so this isn't really needed, right? I would change the users of this public field to use the method instead.

Reply:
It's not needed for us, but `this.id` for `FolderIcon` is something defined by upstream already, so I think I should keep this line (and keep any caller intact) to keep things as similar as possible.